api/scpca_portal/models/original_file.py

```diff
@@ -140,69 +154,79 @@ def purge_deleted_files(
         deletable_files.delete()
         return deletable_file_list
 
-    @staticmethod
-    def is_project_file(s3_key: Path) -> bool:
-        """Checks to see if file is a project data file, and not a library data file."""
-        # project files will not have sample subdirectories
-        return next((True for p in s3_key.parts if common.SAMPLE_ID_PREFIX in p), False)
 
-    @staticmethod
-    def get_relationship_ids(s3_key: Path) -> Tuple:
-        """Parses s3_key and returns project, sample and library ids."""
-        project_id = next(
-            (p for p in s3_key.parts if common.PROJECT_ID_PREFIX in p),
-            None,  # the only file w.o. a project id path part should be the projects metadata file
+@dataclass
+class S3KeyInfo:
+    s3_key: Path
+    project_id: str | None
+    sample_id: str | None
+    library_id_part: str | None
+    is_merged: bool
+    is_bulk: bool
+
+    def __init__(self, s3_key: Path):
+        self.s3_key = s3_key
+        self.project_id = utils.find_first_contained(common.PROJECT_ID_PREFIX, s3_key.parts)
+        self.sample_id = utils.find_first_contained(common.SAMPLE_ID_PREFIX, s3_key.parts)
+        self.library_id_part = utils.find_first_contained(common.LIBRARY_ID_PREFIX, s3_key.parts)
+        self.is_merged = "merged" in s3_key.parts
+        self.is_bulk = "bulk" in s3_key.parts
+
+    @property
+    def library_id(self):
+        if self.library_id_part:
+            return self.library_id_part.split("_")[0]
+        return self.library_id_part
+
+    @property
+    def is_project_file(self):
+        """Project files have project dirs but don't have sample dirs"""
+        return not self.sample_id
```

Comment:
```suggestion
        return self.project_id and not self.sample_id
```

Your comment makes me think we should add this.

Reply:
Like this idea. Had to cast the expression to a bool because `None and <something>` evaluates to None for some reason, which violates the `BooleanField` constraints.